homeassistant/components/roborock/entity.py

```diff
@@ -57,10 +61,14 @@ async def send(
         self,
         command: RoborockCommand | str,
         params: dict[str, Any] | list[Any] | int | None = None,
+        cloud_api: RoborockMqttClientV1 | None = None,
     ) -> dict:
         """Send a command to a vacuum cleaner."""
+        api = self._api
```

Comment:
This seems like an example where inheritence is causing a problem.

My suggestion would be to extract `send` into a helper method that takes a client. The `RoborockEntityV1` can pass in `self.api` and `RoborockCoordinatedEntityV1` can pass in `cloud_api`.

Ideally this would all be transparent to home assistant and just handled in the python-roborock library, but i realize that is a bigger change.